### PR TITLE
feat: log response from server and pretty log

### PR DESCRIPTION
**Pull request by PrashantRaj18198**
## Desc

Logs server response on prehook in case of failure. Also, some prettification.

> In case of failure

image

> In case of success

image

### CODE REVIEW

File: tools/github/hooks/prerun.sh [hunk 1]
```diff
@@ -26,13 +28,29 @@ cat <<EOF > warpbuild_body.json
...
+  -O warpbuild_response.json --post-file=warpbuild_body.json \
+  "$WARPBUILD_HOST_URL/api/v1/job" || exit_code=$? || true
+
+if [ -n "$exit_code" ]; then
+    echo "Failed to send request to warpbuild. Logging response. Exiting..."
+    # check if jq is installed if so then pretty print the json response
+    if ! command -v jq &> /dev/null; then
+        cat warpbuild_response.json
+    else
+        jq . warpbuild_response.json
```
Comment (guptaankit015) [inline tools/github/hooks/prerun.sh:49]: if we are printing complete backend error, then we should also structure our error messages/description sent from backend in this api

**Review by guptaankit015 — COMMENTED** [review]
LGTM

File: tools/github/hooks/prerun.sh (hunk 1 above)
Comment (PrashantRaj18198) [reply tools/github/hooks/prerun.sh:49]: That is structured, right?

File: tools/github/hooks/prerun.sh [hunk 2]
```diff
@@ -1,14 +1,16 @@
 #!/bin/bash
 
-export TEST_VAR="Hello from $WARPBUILD_RUNNER_SET_ID's pre processor!"
-echo "$TEST_VAR"
+# this only used for script debug (uncomment if needed)
+# echo "Script ID: 3"
+echo "Running prehook for warpbuild runner instance '$WARPBUILD_RUNNER_SET_ID'..."
```
Comment (abhijit-hota) [inline tools/github/hooks/prerun.sh:5]:
```suggestion
echo "Running prehook for WarpBuild runner instance '$RUNNER_NAME'..."
```

File: tools/github/hooks/prerun.sh [hunk 3]
```diff
@@ -26,13 +28,29 @@ cat <<EOF > warpbuild_body.json
 }
 EOF
 
+echo -e "\nMaking a request to warpbuild..."
```
Comment (abhijit-hota) [inline tools/github/hooks/prerun.sh:31]:
```suggestion
echo -e "\nMaking a request to WarpBuild..."
```

File: tools/github/hooks/prerun.sh [hunk 4]
```diff
@@ -26,13 +28,29 @@ cat <<EOF > warpbuild_body.json
...
+        jq . warpbuild_response.json
+    fi
+    exit 1
+fi
+
+rm warpbuild_body.json
 
-rm warpbuild_body.json
\ No newline at end of file
+echo -e "\nPrehook for warpbuild runner instance '$WARPBUILD_RUNNER_SET_ID' completed succesfully."
```
Comment (abhijit-hota) [inline tools/github/hooks/prerun.sh:56]:
```suggestion
echo -e "\nPrehook for WarpBuild runner instance '$RUNNER_NAME' completed succesfully."
```

**Review by abhijit-hota — APPROVED** [review]

**Comment by abhijit-hota** [comment]
Tested again: https://github.com/WarpBuilds/kitchen-sink/actions/runs/9127626651/job/25098295163

**Pull request merged**